elastic/eui
# [EuiSuperDatePicker][EuiAutoRefresh][EuiRefreshInterval] Allow controlling the rendered unit

## Summary

closes #5571 and #5752

![refresh_interval](https://github.com/elastic/eui/assets/549407/41fd1e01-2dc6-44af-b385-5b7583592557)

This PR adds a `refreshIntervalUnits` prop to **EuiSuperDatePicker** and an `intervalUnits` prop to **EuiAutoRefresh** and **EuiRefreshInterval**. It also updates the `onRefreshChange` callback of all 3 components to pass back a new `intervalUnits` key.

This allows consumers to receive and set their own interval unit state based on user selection (or even ignoring user selection, if they prefer to always coerce a certain unit) instead of the default behavior, which rounds the rendered unit up to the nearest largest unit (e.g. 90s = 2m). If users still want this default unit rounding, they can simply omit passing an `intervalUnits` prop.

## QA

- Go to https://eui.elastic.co/pr_7501/#/templates/super-date-picker#auto-refresh
- Click the **Quick select menu** button in the form prepend
- Check the **Refresh every** toggle
- [x] Type a large number in (e.g. 1000) and confirm that the appended auto refresh interval outside the popover rounds up to minutes
- Uncheck the toggle for rounding units
- [x] Confirm the appended interval matches the unit selected in the dropdown
- [x] Change the unit selection and confirm the appended interval unit updates accordingly, but not the number 
- Go to https://eui.elastic.co/pr_7501/#/forms/auto-refresh
- For each of the 3 playgrounds on the page:
  - Open the playground and toggle the `isPaused` prop to false
  - [x] Set `refreshInterval` to 100000 and `intervalUnits` to `m` and confirm the corresponding number and unit are rendered as expected

### General checklist

- Browser QA
- Docs site QA
    - [x] Updated **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)** example
- Code quality checklist
- Release checklist
- Designer checklist - N/A
